lib/topic_view.rb

```diff
@@ -198,11 +198,16 @@ def prev_page
   end
 
   def next_page
-    @next_page ||= begin
-      if last_post && highest_post_number && (highest_post_number > last_post.post_number)
-        @page + 1
+    return @next_page if defined?(@next_page)
```

Comment:
`@next_page` can be `nil` and previously we were not memoizing that value properly.